crates/data_components/src/duckdb.rs

```diff
@@ -137,18 +138,13 @@ fn to_datafusion_error(error: Error) -> DataFusionError {
 #[derive(Clone)]
 pub struct DuckDB {
     table_name: String,
-    schema: SchemaRef,
```

Comment:
What is the rationale for removing schema here?

Reply:
The only place where the schema field is really needed is in the original create_table function, specifically in the overwrite case. However, previously we are mixing the use of creating new table, recreating table, and overwrite old table within the create_table function. 

In creating a new table and recreating table cases, schema exists near create_table. It would make more sense to pass it as create_table input, instead of making it a field for duckdb. (Line 115: https://github.com/spiceai/spiceai/blob/639df26a93a714f32455e16387b7f85bd3ce652d/crates/data_components/src/duckdb.rs#L115)

The overwrite old table is what only requires the schema field. The previous way we do it is by dropping the old table, build an empty record batch with old table schema retrieved from the duckdb field, and recreate the table with the empty record from it. While what we can do is simply delete all of the rows in the old table, which doesn’t need a schema. (Line 119: https://github.com/spiceai/spiceai/blob/639df26a93a714f32455e16387b7f85bd3ce652d/crates/data_components/src/duckdb/write.rs#L119)

Reply:
I think logically it makes sense for this struct to have the schema up front, even if its technically only needed at the create_table level.

But if you think it makes more sense to remove it, then we should make it consistent with the other data components as well.

Reply:
Yeah, I think it logically makes sense if the schema is defined and not expected to change per DuckDB initialization. I’ll keep the schema area if so. I was referring to postgres implementation when changing duckdb and postgres doesn't keep a schema up front, so I was not sure whether to keep schema or not for DuckDB.

But I think we should keep the delete_all_table_data added, it makes the rewrite more efficient than the previous create_table cuz we only connected to DuckDB and executed the statement once. 

Would you share your thoughts on these? Thanks!

Reply:
Yes, the `delete_all_table_data` is good 👍

Reply:
And the Postgres one should be changed to match this as well. That was an oversight on my part.

Reply:
DuckDB Schema parameter added back